### ISSUE TITLE
ControllerBase Problem() does not return correct HTTP Status Code header

### ISSUE THREAD

**Issue by vmcbaptista** [aspnet/AspNetCore#15103]
### Describe the bug
When using Problem() from ControllerBase to return a response body of type ProblemDetails, the status code is not reflected in HTTP Status Code header.

### To Reproduce
1. Using this version of ASP.NET Core 3.0
2. On an API controller method `return Problem(detail: "ErrorDetail", statusCode: 400);`
4. The response is:
```
$ curl --location --request GET "http://127.0.0.1:5000/api/error" -k -i
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   158    0   158    0     0   2925      0 --:--:-- --:--:-- --:--:--  2925

HTTP/1.1 200 OK
Date: Thu, 17 Oct 2019 08:22:46 GMT
Content-Type: application/problem+json; charset=utf-8
Server: Kestrel
Transfer-Encoding: chunked

{"type":"https://tools.ietf.org/html/rfc7231#section-6.5.1","title":"Bad Request","status":400,"detail":"ErrorDetail","traceId":"|1452e238-442aa39a3bcece98."}
```

### Expected behavior
I expect to receive a response with HTTP Status code, matching the status described in ProblemDetails object.

I believe this could be fixed if we change the following line 
https://github.com/aspnet/AspNetCore/blob/bd65275148abc9b07a3b59797a88d485341152bf/src/Mvc/Mvc.Core/src/ControllerBase.cs#L1897

to something like
`return new ObjectResult(problemDetails) {StatusCode = statusCode ?? 500}`

**Comment by Kahbazi** [comment]
This issue has been fixed via https://github.com/aspnet/AspNetCore/pull/14672 for release/3.1 but has not been merged to master yet

**Comment by javiercn** [comment]
This has been fixed in 3.1
